doc/Developer-Guide.md

```diff
@@ -0,0 +1,453 @@
+
+Note, JetBrains produts that requires use activation code for offline usage should have version at least **2018.1 or later**.
+
+
+
+## Quick start
+
+This section describes the several ways how to build images with specific JetBrains product.
+
+
+
+##### Build IntelliJ Idea Community Edition
+
+To build IntelliJ Idea Community Edition it is enough to call the following command inside repository folder:
+
+```
+$ docker build -t idea-ic .
+```
+
+> Optionally, you can provide build argument:
+>
+> ```
+> PRODUCT_NAME=ideaIC
+> ```
+>
+>  but it is uneccessary. If no `--build-arg` was passed, `ideaIC` is used instead.
+>
+> The whole command will look like:
+>
+> ```
+> $ docker build -t idea-ic --build-arg PRODUCT_NAME=ideaIC .
+> ```
+
+This command will build image with **2020.2.3** version by default. Then it is enough to tag result image and push tagged image to your repository:
+
+```
+$ docker tag idea-ic:latest <username>/idea-ic:latest
+$ docker push <username>/idea-ic:latest
+```
+
+Now this image can be used as Che Editor. To achive this there are two yaml configurations needs to be created:
+
+- `workspace.yaml` – workspace configuration. Do not forged to provide correct url to the `meta.yaml` file:
+
+  ```
+  metadata:
+    name: che-ideaic
+  components:
+    - type: cheEditor
+      reference: '<url for the meta.yaml goes here>'
+      alias: theia-editor
+  apiVersion: 1.0.0
+  ```
+
+- `meta.yaml` – Che Editor configuration. Do not forget to replace `<username>` with the user name where image is pushed to:
+
+  ```
+  apiVersion: v2
+  publisher: <username>
+  name: intellij-NOVNC
+  version: 2020.2.3
+  type: Che Editor
+  displayName:  IntelliJ IDEA Community Edition
+  title:  IntelliJ IDEA Community Edition (in browser using noVNC) as editor for Eclipse Che
+  description:  IntelliJ IDEA Community Edition running on the Web with noVNC
+  icon: https://resources.jetbrains.com/storage/products/intellij-idea/img/meta/intellij-idea_logo_300x300.png
+  category: Editor
+  repository: https://github.com/che-incubator/che-editor-intellij-community
+  firstPublicationDate: "2020-10-27"
+  spec:
+    endpoints:
+    # hack (use dirigible endpoint name as it is part of liveness probes)
+     -  name: "dirigible"
+        public: true
+        targetPort: 8080
+        attributes:
+          protocol: http
+          type: ide
+          path: /vnc.html?resize=remote&autoconnect=true&reconnect=true
+    containers:
+     - name: intellij-novnc
+       image: "<username>/idea-ic:latest"
+       mountSources: true
+       volumes:
+           - mountPath: "/JetBrains/ideaIC"
+             name: idea-configuration
+       ports:
+           - exposedPort: 8080
+       memoryLimit: "2048M"
+  
+  ```
+
+
+
+As an example showed in the picture: 
+
+![create-workspace-from-devfile](https://raw.githubusercontent.com/che-incubator/che-editor-intellij-community/master/doc/images/create-workspace-from-devfile.jpg)
```

Comment:
images should go in a separate branch to not make source code too big
https://medium.com/@minamimunakata/how-to-store-images-for-use-in-readme-md-on-github-9fb54256e951

Reply:
Moved images to branch `media`. It was created with option `--orphan`, so it's history is clear.